core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessage.java

```diff
@@ -183,6 +189,10 @@ protected void subscribe0(EventExecutor executor, SubscriptionOption[] options)
                 } else {
                     executor.execute(() -> collectAll(collectingFuture, executor, withPooledObjects, false));
                 }
+            } else {
+                // We don't need to invoke onSubscribe() for NoopSubscriber.
+                invokedOnSubscribe = true;
+                notifySubscriber();
```

Comment:
Do we need to call this method here? 🤔

Reply:
Even though a stream is open, we need to drain `queue` so that `onRemoval()` should be called immediately.

Reply:
What I meant was that if this thread fails to set the state as `State.CLEANUP`, then there might be another thread that handles the elements.
So I was wondering in which case we need to call this method. 🤔

Reply:
Ah I realized that the `State` can be `State.OPEN` so we need to call that. 😅